postprocessing/visualization/receiver/src/Waveform.py

```diff
@@ -67,4 +70,8 @@ def normalize(self):
     for name, wf in self.waveforms.items():
       if self.norm[name] > numpy.finfo(float).eps:
         self.waveforms[name] = wf / self.norm[name]
-    
+
+  def differentiate(self):
+    dt = self.time[1] - self.time[0]
+    for name, wf in self.waveforms.items():
+      self.waveforms[name] = numpy.diff(wf, prepend=0.0) / dt
```

Comment:
Maybe more accurate: https://numpy.org/doc/stable/reference/generated/numpy.gradient.html#numpy.gradient

Reply:
I forgot one file, thanks for noticing.